simulation/traffic_simulator/src/entity/entity_base.cpp

```diff
@@ -713,5 +713,149 @@ auto EntityBase::updateTraveledDistance(const double step_time) -> double
   return traveled_distance_;
 }
 
+/**
+ * @brief Calculates the distance between the entity and the target lanelet pose.
+ *
+ * This function calculates the longitudinal distance between the entity's current lanelet pose
+ * and the target lanelet pose. It calculates the distance in both directions (forward and backward) since sometimes the entity may have already passed the target lanelet pose.
+ *
+ * @param target_lanelet_pose The target lanelet pose to calculate the distance to.
+ * @return The distance between the entity and the target lanelet pose.
+ * @throws SemanticError If the entity's lanelet pose is not available.
+ * @throws SimulationError If the distance between the entity and the target lanelet pose cannot be calculated.
+ */
+auto EntityBase::getDistanceToTargetLaneletPose(
+  const CanonicalizedLaneletPose & target_lanelet_pose) -> double
+{
+  const auto entity_lanelet_pose_ = getLaneletPose();
+  const auto target_lanelet_pose_ = target_lanelet_pose;
+
+  if (!entity_lanelet_pose_) {
+    THROW_SEMANTIC_ERROR(
+      "Failed to get entity's lanelet pose. Please check entity lanelet pose exists.");
+  }
+
+  const auto entity_distance_to_target = hdmap_utils_ptr_->getLongitudinalDistance(
+    static_cast<LaneletPose>(entity_lanelet_pose_.value()),
+    static_cast<LaneletPose>(target_lanelet_pose_));
+
+  const auto revert_entity_distance_to_target = hdmap_utils_ptr_->getLongitudinalDistance(
+    static_cast<LaneletPose>(target_lanelet_pose_),
+    static_cast<LaneletPose>(entity_lanelet_pose_.value()));
+
+  if (!revert_entity_distance_to_target.has_value() && !entity_distance_to_target.has_value()) {
+    THROW_SIMULATION_ERROR("Failed to get distance between entity and target lanelet pose.");
+  }
+
+  if (!revert_entity_distance_to_target.has_value()) {
+    return entity_distance_to_target.value();
+  }
+
+  if (!entity_distance_to_target.has_value()) {
+    return revert_entity_distance_to_target.value();
+  }
+
+  if (revert_entity_distance_to_target.value() < entity_distance_to_target.value()) {
+    return revert_entity_distance_to_target.value();
+  } else {
+    return entity_distance_to_target.value();
+  }
+}
+
+bool EntityBase::reachPosition(const std::string & target_name, const double tolerance) const
+{
+  return reachPosition(other_status_.find(target_name)->second.getMapPose(), tolerance);
+}
+
+bool EntityBase::reachPosition(
+  const geometry_msgs::msg::Pose & target_pose, const double tolerance) const
+{
+  return math::geometry::getDistance(getMapPose(), target_pose) < tolerance;
+}
+
+bool EntityBase::reachPosition(
+  const CanonicalizedLaneletPose & lanelet_pose, const double tolerance) const
+{
+  return reachPosition(static_cast<geometry_msgs::msg::Pose>(lanelet_pose), tolerance);
+}
```

Comment:
Please use [this functions.](https://github.com/tier4/scenario_simulator_v2/blob/45d42a79d92c370387749ad16c10665deb42e02c/simulation/traffic_simulator/include/traffic_simulator/utils/distance.hpp#L36-L40)